pyiron/atomistics/structure/atoms.py

```diff
@@ -1213,17 +1227,26 @@ def plot3d(self, show_cell=True, show_axes=True, camera='perspective', spacefill
         view = nglview.NGLWidget(struct)
 
         if spacefill:
-            if color_scheme is None and scalar_field is not None:
-                color_scheme = 'occupancy'
-            elif color_scheme is None:
-                color_scheme = 'element'
-            # TODO: Raise a warning if `color_scheme` and `scalar_field` are both given.
-
-            for elem, num in set(list(zip(parent_basis.get_chemical_symbols(), parent_basis.get_atomic_numbers()))):
-                view.add_spacefill(selection='#' + elem,
-                                   radius_type='vdw',
-                                   color_scheme=color_scheme,
-                                   radius=particle_size * (0.2 + 0.1 * np.sqrt(num)))
+            elemental_data_list = list(zip(elements, atomic_numbers))
+
+            if scalar_field is not None:
+                assert(color_scheme is None)
```

Comment:
Apparently runtime `assert` statements should be minimized [(see here)](http://web.mit.edu/6.005/www/fa16/classes/08-avoiding-debugging/#what_not_to_assert). So maybe an `if` statement followed by raising a `ValueError` may be better?

Reply:
Noted. I actually replaced them with warnings instead.